NEW_PROJECTS.md

```diff
@@ -0,0 +1,70 @@
+# Creating a new SPIFFE Project
```

Comment:
Suggest adding an `## Introduction` section.

a brief section at the beginning that summarizes what SPIFFE is and why someone might be interested in contributing. This would help set the context for the reader.